### PR TITLE
WT-76 - Detect type of chain on activation

### PR DESCRIPTION
**Pull request by AgusVelez5**
**Type of MR:**

- [x] Feature
- [ ] Fix
- [ ] HotFix

**Small description:**

- Add detectLegacyChain

**Aditional observations/considerations:**
...

# Checklist:

- [x] My code follows the styleguides of the project (successfully pass linters)
- [x] I tested my code.
- [x] I commented/docummented my code in a way that complex parts are explained.
- [ ] If required, i updated project's README file.
- [x] I added/updated tests that tests my changes in code.
- [x] All previous and new tests pass.
- [ ] I commit any new dependency.
- [ ] I updated UML diagram in confluence if any model was modified in this MR.

### CODE REVIEW

File: src/managers/ContractManager.ts [hunk 1]
```diff
@@ -322,12 +323,36 @@ export class ContractManager {
...
+    }
+
     if (iFunction.inputs.length !== params.length)
       throw errorTypes.INVALID_PARAMETER_COUNT(
         iFunction.inputs.length,
         params.length
       )
 
+    // Remove gas price
+    if (
```
[the remark this answers is not in the document]
Comment (alfarkas) [reply src/managers/ContractManager.ts:349]: That makes sense, but what is the criteria in that case to decide to leave the parameters for eip1559 and remove the legacy one if it was also provided? I think that in that case is best to let the user decide. Provide one or the other, not both.

[the remark this answers is not in the document]
Comment (AgusVelez5) [reply src/managers/ContractManager.ts:349]: I think we can take two paths:
 - Throw an error when it sends both parameters
 - Log a warning, it seems to me that both values can be passed to ethers, right?

[the remark this answers is not in the document]
Comment (alfarkas) [reply src/managers/ContractManager.ts:349]: Yeah, i think we can go for the second one too. Something like "`gasPrice was provided, maxPriorityFeePerGas and maxFeePerGas will be ignored`"?

**Comment by alfarkas** [comment]
Change destination branch to develop.

**Review by alfarkas — DISMISSED** [review]

**Review by fedeaviles — APPROVED** [review]
🚀

**Review by alfarkas — DISMISSED** [review]

**Review by alfarkas — APPROVED** [review]

**Review by mauricio-b-r — APPROVED** [review]

**Review by fedeaviles — APPROVED** [review]

**Pull request merged**